pkg/routing/adapter/filter/adapter_test.go

```diff
@@ -0,0 +1,319 @@
+		filters: map[string]outputs{
+			tNS + "/filter-1": {
+				wantError: false,
+				pass:      false,
+			},
+			tNS + "/filter-2": {
+				wantError: false,
+				pass:      false,
+			},
+		},
+	},
+	"failingEvent": {
+		payload: `{}`,
+		filters: map[string]outputs{
+			tNS + "/malformed": {
+				wantError: true,
+				pass:      false,
+			},
+			"wrongns/filter": {
+				wantError: true,
+				pass:      false,
+			},
+			"/filter": {
+				wantError: true,
+				pass:      false,
+			},
+			"bad/request/filter": {
+				wantError: true,
+				pass:      false,
+			},
+			tNS + "/missingfilter": {
+				wantError: true,
+				pass:      false,
+			},
+		},
+	},
+}
+
+func TestAdapter(t *testing.T) {
+	ip, port := testServerSocket(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	responses, sinkAddr := setupSink(t, ctx)
+	ctx = setupFakeInformers(t, ctx, sinkAddr.String())
+
+	h := newHandler(t, ctx, port)
+	go func() {
+		if err := h.Start(ctx); err != nil {
+			assert.FailNow(t, "could not start test adapter", err)
+		}
+	}()
+	<-h.receiver.Ready
+
+	for name, tc := range tCases {
+		for filter, outputs := range tc.filters {
+			t.Run(fmt.Sprintf("%s(%s)", filter, name), func(t *testing.T) {
+				filterAddr := fmt.Sprintf("http://%s:%d/%s", ip, port, filter)
+
+				ce := newCloudEvent(t, tc.payload)
+				filterResponse := sendCE(t, &ce, filterAddr)
+				assert.Equal(t, outputs.wantError, cloudevents.IsNACK(filterResponse), "Filter didn't report bad request")
+
+				select {
+				case sinkResponse := <-responses:
+					if outputs.pass {
+						assert.Equal(t, tc.payload, sinkResponse)
+					} else {
+						t.Error("Unexpected event passed the filter")
+					}
+				default:
+					if outputs.pass {
+						t.Errorf("Expected event didn't pass the filter")
+					}
+				}
+			})
+		}
+	}
+}
+
+func newCloudEvent(t *testing.T, data string) cloudevents.Event {
+	event := cloudevents.NewEvent()
+	event.SetID(tCloudEventID)
+	event.SetType(tCloudEventType)
+	event.SetSource(tCloudEventSource)
+	err := event.SetData(cloudevents.ApplicationJSON, []byte(data))
+	require.NoError(t, err)
+	return event
+}
+
+func sendCE(t *testing.T, event *cloudevents.Event, sink string) protocol.Result {
+	ctx := cloudevents.ContextWithTarget(context.Background(), sink)
+	c, err := cloudevents.NewClientHTTP()
+	require.NoError(t, err)
+
+	return c.Send(ctx, *event)
+}
+
+func setupFakeInformers(t *testing.T, ctx context.Context, sinkURI string) context.Context { //nolint:staticcheck
+	var filters []runtime.Object
+	for _, f := range tFilters {
+		filters = append(filters, newFilter(t, f.key, f.expression, sinkURI))
+	}
+
+	injection.Fake.RegisterClient(func(ctx context.Context, _ *rest.Config) context.Context {
+		ctx, _ = fakeinjectionclient.With(ctx, filters...)
+		return ctx
+	})
+
+	var infs []controller.Informer
+	ctx, infs = rectesting.SetupFakeContext(t) //nolint:staticcheck
```

Comment:
The warning was legit ⚠️
You're overriding the context instead of infusing it

Reply:
The fix is to replace this call by that call:
https://github.com/knative/pkg/blob/5ac4a75f34884a2adc4a77ce1a6cb05cc29d897c/reconciler/testing/context.go#L57

Reply:
Thanks for the hint!